Automattic/wp-calypso
# Editor: Refactor page templates to use Redux state

This pull request seeks to deprecate Flux-based page templates state in favor of global Redux-based state introduced in #7174. It introduces a new `<QueryPageTemplates />` query component, refactoring the `<EditorPageTemplates />` component to make use of it. Since this is the only area of the application using Flux-based state and the `<PageTemplatesData />` component, both of these were removed.

__Testing instructions:__

Verify that page template options are shown and can be managed, especially with the following considerations:

- Page templates should only be shown when on the [page editor](http://calypso.localhost:3000/page), not for other post types
- Page templates should only be shown when two or more options exist
- Page templates should be updated when navigating back to the editor after having changed the theme (see #5387, #5785, cc @jeremeylduvall)


Test live: https://calypso.live/?branch=update/editor-page-templates
